contracts/DAORegistryScheme.sol

```diff
@@ -0,0 +1,181 @@
+
+    // A mapping from hashes to parameters (use to store a particular configuration on the controller)
+    struct Parameters {
+        bytes32 voteParams;
+        IntVoteInterface intVote;
+        DAORegistry daoRegistry;
+    }
+
+    mapping(bytes32=>Parameters) public parameters;
+
+    /**
+     * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+     * @param _proposalId the ID of the voting in the voting machine
+     * @param _param a parameter of the voting result, 1 yes and 2 is no.
+     */
+    function executeProposal(bytes32 _proposalId, int256 _param) external onlyVotingMachine(_proposalId) returns(bool) {
+        Avatar avatar = proposalsInfo[msg.sender][_proposalId].avatar;
+        DAORegistryProposal memory proposal = organizationsProposals[address(avatar)][_proposalId];
+        delete organizationsProposals[address(avatar)][_proposalId];
+        emit ProposalDeleted(address(avatar), _proposalId);
+
+        if (_param == 1) {
+            Parameters memory params = parameters[getParametersFromController(avatar)];
+            bytes memory genericCallReturnValue;
+            bool success;
+            ControllerInterface controller = ControllerInterface(avatar.owner());
+            (success, genericCallReturnValue) =
+                controller.genericCall(address(params.daoRegistry), proposal.callData, avatar, proposal.value);
+            require(success, "proposal external call cannot be executed");
+            emit ProposalExecuted(address(avatar), _proposalId, _param, genericCallReturnValue);
+        }
+
+        return true;
+    }
+
+    /**
+    * @dev hash the parameters, save them if necessary, and return the hash value
+    */
+    function setParameters(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        DAORegistry _daoRegistry
+    ) public returns(bytes32)
+    {
+        bytes32 paramsHash = getParametersHash(_voteParams, _intVote, _daoRegistry);
+        parameters[paramsHash].voteParams = _voteParams;
+        parameters[paramsHash].intVote = _intVote;
+        parameters[paramsHash].daoRegistry = _daoRegistry;
+        return paramsHash;
+    }
+
+    function getParametersHash(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        DAORegistry _daoRegistry
+    ) public pure returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(_voteParams, _intVote, address(_daoRegistry)));
+    }
+
+    /**
+    * @dev create a proposal to register a DAO in the registry
+    * @param _avatar the address of the organization owning the registry
+    * @param _proposedAvatarName the name of the organization we want to add to the registry
+    * @param _proposedAvatar the organization we want to add to the registry
+    * @param _value value(ETH) to transfer with the call
+    * @return a proposal Id
+    */
+    function proposeToAddDAO(
+        Avatar _avatar,
```

Comment:
why cast his as an `Avatar` if you are then recasting it to an address everywhere below?